### DISCUSSION TITLE
Tjoy - Spadnext doesn't 'talk'

### DISCUSSION THREAD

**Discussion by hrcr117** [mpaperno/TJoy#5] [Q&A]
Hi, I want to use TJoy as a tool for double way communication between spadnext and touchportal (through vjoy of course).  My logic is: push tp button -> vjoy button pushed -> spad recognise it -> msfs event; then: spad recognise that msfs even happend -> push other vjoy button -> (problem is here here) TJoy recognise pushed vjoy button -> change button state in TP to on. The problem is, when Tjoy tries to read vjoy pressed button its overrides spadnext pushes. Is there any way to solve it with Tjoy?

**Comment by mpaperno** [comment]
Hi,

I don't really understand this last part:
"when Tjoy tries to read vjoy pressed button its overrides spadnext pushes."

TJoy "reading" the vJoy values/inputs shouldn't be affecting/changing the inputs.

I don't know how SPAD handles vJoy, but only one program/app at a time can be controlling a vJoy device (this is a limitation of the driver).  If TJoy and SPAD are both trying to control/use the same vJoy device, that probably won't work.  IIRC that would apply even if TJoy is only "listening" for input changes (to apply button visuals or whatnot), though I'm actually not 100% sure on that.

You could perhaps try using separate vJoy devices for TJoy and SPAD, if that's practical for your needs.  That would allow independent controls at least -- not sure it would solve the feedback issue.

You're aware of the MSFS TP plugin as well, yes?

Cheers,
-Max
